app/static/js/manageDepartments.js

```diff
@@ -7,8 +7,88 @@ $(document).ready( function(){
     x.DataTable({
         pageLength: 25
     });
+
+    attachModalToDepartment()
+    $('#departmentsTable').on('draw.dt', function() {
+      attachModalToDepartment()
+    })
+    $('#manageDepartmentSupervisorModal').on('hidden.bs.modal', function() {
+      clearDropdown()
+    })
 });
 
+
+function attachModalToDepartment() {
+    $('.supervisorDeptModal').off('click')
+    $('.supervisorDeptModal').on('click', function() {
+      $('#manageSupervisorContent .modal-header .department-name')
+                .replaceWith(`<h2 class='department-name' id="departmentModalSelect" data-department-id="${this.id}">
+                                    ${$(`#${this.id}`).html()}
+                              </h2>`)
+      getSupervisorsInDepartment(this.id)
+    })
+}
+
+
+$("#supervisorModalSelect").on('change', async function() {
+  let supervisor = $('#supervisorModalSelect :selected').val()
+  let department = $('#departmentModalSelect').data('department-id')
+  let data = {"supervisor": supervisor, "department": department}
+  await addSupervisorToDepartment(data)
+  getSupervisorsInDepartment(department)
+})
+
+  
+
+function getSupervisorsInDepartment(departmentID) {
+    $.ajax({
+    method: "GET",
+    url: `/admin/manageDepartments/${departmentID}`,
+    success: function(departmentsAndSupervisors) {
+      let currentDepartment=departmentsAndSupervisors[0]
+      let supervisors=departmentsAndSupervisors[1]
+      
+      let supervisorContent = '<div class="changing-content">'
+      for (let i=0; i<supervisors.length; i++) {
+        let supervisorFirstName= supervisors[i]['preferred_name'] ? supervisors[i]['preferred_name'] : supervisors[i]['legal_name']
+        supervisorContent += (`<span class="row">
+                                <div class="col-xs-10">${supervisors[i]['ID']} ${supervisorFirstName} ${supervisors[i]['LAST_NAME']}</div>
+                                <div class='btn btn-danger col-auto removeSupervisorFromDepartment' 
+                                  data-supervisor="${supervisors[i]['ID']}" 
+                                  data-department="${currentDepartment['departmentID']}" 
+                                  id="${supervisors[i]['ID']}-${currentDepartment['departmentID']}">Remove</div>
+                               </span>`)}
+      supervisorContent += ("</div>")
+      $('#manageSupervisorContent .modal-body .changing-content').replaceWith(supervisorContent)
+      
+      $('#manageDepartmentSupervisorModal').modal('show')
+      $('.removeSupervisorFromDepartment').on('click', removeSupervisorFromDepartment)
+    }
+    })
+  }
+
+function removeSupervisorFromDepartment () {
+  let department = $(`#${this.id}`).data('department')
+  let supervisor = $(`#${this.id}`).data('supervisor')
+  let data = {"supervisor": supervisor, "department": department}
```

Comment:
As I mentioned in the python code segment, I'd consider renaming the keys here as such for clarity about what the data holds:
`supervisor` --> `supervisorId`
`department` --> `departmentId`